File: Songhay.Dashboard/ClientApp/projects/songhay-player-video/src/lib/you-tube/models/you-tube-scalars.ts
/**
 * centralizes magic strings for YouTube services, components, etc.
 *
 * @export
 * @class YouTubeScalars
 */
export class YouTubeScalars {
    /**
     * magic string
     *
     * @static
     * @memberof YouTubeScalars
     */
    static rxYouTubeApiRootUri =
        'https://songhay-system-player.azurewebsites.net/api/Player/v1/video/youtube/';

    /**
     * magic string
     *
     * @static
     * @memberof YouTubeScalars
     */
    static rxYouTubeApiPlaylistPath = 'playlist/uploads/';

    /**
     * magic string
     *
     * @static
     * @memberof YouTubeScalars
     */
    static rxYouTubeApiPlaylistsPath = 'playlists/';

    /**
     * magic string
     *
     * @static
     * @memberof YouTubeScalars
     */
    static rxYouTubeApiPlaylistsIndexPath = 'playlist/index/';

    /**
     * magic string
     *
     * @static
     * @memberof YouTubeScalars
     */
    static rxYouTubeApiVideosPath = 'videos/';

    /**
     * magic string
     *
     * @static
     * @memberof YouTubeScalars
     */
    static rxYouTubeWatchRootUri = 'https://www.youtube.com/watch?v=';

    static setupForJasmine(): void {
        YouTubeScalars.rxYouTubeApiRootUri = './base/src/assets/data/';
    }
}

File: Songhay.Player.YouTube/Models/YouTubeScalars.fs
namespace Songhay.Player.YouTube.Models

module YouTubeScalars =

    /// <summary>
    /// base URI representing the b-roll endpoint
    /// for YouTube curation
    /// </summary>
    [<Literal>]
    let YouTubeApiRootUri = "https://songhay-system-player.azurewebsites.net/api/Player/v1/video/youtube/"

    /// <summary>
    /// URI path representing a YouTube channel
    /// and/or YouTube `uploads`
    /// curated as a `playlist`
    /// </summary>
    [<Literal>]
    let YouTubeApiPlaylistPath = "playlist/uploads/"

    /// <summary>
    /// URI path representing sets of YouTube `uploads`
    /// curated as `playlists`
    /// </summary>
    [<Literal>]
    let YouTubeApiPlaylistsPath = "playlists/"

    /// <summary>
    /// URI path representing indices of curated as `playlists`
    /// in @type {GenericWebIndex} format
    /// </summary>
    [<Literal>]
    let YouTubeApiPlaylistsIndexPath = "playlist/index/"

    /// <summary>
    /// URI path representing curated `videos`
    /// or YouTube `items` for a gen-web
    /// </summary>
    [<Literal>]
    let YouTubeApiVideosPath = "videos/"

    /// <summary>
    /// base URI representing YouTube channel browsing
    /// </summary>
    [<Literal>]
    let YouTubeChannelRootUri = "https://www.youtube.com/channel/"

    /// <summary>
    /// base URI representing YouTube item watching
    /// </summary>
    [<Literal>]
    let YouTubeWatchRootUri = "https://www.youtube.com/watch?v="
